sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java

```diff
@@ -236,8 +236,15 @@ public UnboundedJmsSource(
     public List<UnboundedJmsSource> generateInitialSplits(
         int desiredNumSplits, PipelineOptions options) throws Exception {
       List<UnboundedJmsSource> sources = new ArrayList<>();
-      for (int i = 0; i < desiredNumSplits; i++) {
+      if (topic != null) {
+        // in the case of a topic, we create a single source, so an unique subscriber, to avoid
```

Comment:
I might be totally uninformed, but is it possible to make a queue out of a topic? Said differently, is it possible to separate a given topic into multiple queues?